sdk/apis/catalog/v1alpha1/collection_types.go

```diff
@@ -0,0 +1,94 @@
+/*
+Copyright 2025 The Kube Bind Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package v1alpha1
+
+import (
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+
+	conditionsapi "github.com/kube-bind/kube-bind/sdk/apis/third_party/conditions/apis/conditions/v1alpha1"
+)
+
+// Collection groups multiple Modules into a logical group. This functions as a folder in the UI.
+//
+// +crd
+// +genclient
+// +k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object
+// +kubebuilder:resource:scope=Namespaced,categories=kube-bindings
+// +kubebuilder:subresource:status
+// +kubebuilder:printcolumn:name="Description",type="string",JSONPath=`.spec.description`
+// +kubebuilder:printcolumn:name="Modules",type="integer",JSONPath=`.spec.modules[*]`
+// +kubebuilder:printcolumn:name="Age",type="date",JSONPath=`.metadata.creationTimestamp`,priority=0
+type Collection struct {
+	metav1.TypeMeta   `json:",inline"`
+	metav1.ObjectMeta `json:"metadata,omitempty"`
+
+	// spec specifies the collection.
+	// +required
+	// +kubebuilder:validation:Required
+	Spec CollectionSpec `json:"spec"`
+
+	// status contains reconciliation information for the collection.
+	Status CollectionStatus `json:"status,omitempty"`
+}
+
+func (in *Collection) GetConditions() conditionsapi.Conditions {
+	return in.Status.Conditions
+}
+
+func (in *Collection) SetConditions(conditions conditionsapi.Conditions) {
+	in.Status.Conditions = conditions
+}
+
+// CollectionSpec defines the desired state of Collection.
+type CollectionSpec struct {
+	// description is a human readable description of this collection.
+	//
+	// +optional
+	Description string `json:"description,omitempty"`
+
+	// modules is a list of module references that are part of this collection.
+	//
+	// +required
+	// +kubebuilder:validation:Required
+	// +kubebuilder:validation:MinItems=1
```

Comment:
I don't think we need to require a minimum amount of items, or? It doesn't make much of a difference and it would be a breaking change if a match expression/label selector is added next to the modules.

Reply:
It's fine, let's merge it, and I'm gonna work on actually using it. And we can relax these once time comes